data/wildfly-runner/pom.xml

```diff
@@ -0,0 +1,309 @@
+                    <reportNameSuffix>core</reportNameSuffix>
+                    <systemPropertyVariables>
+                        <jimage.dir>target/jimage</jimage.dir>
+                        <jakarta.tck.profile>core</jakarta.tck.profile>
+                        <signature.sigTestClasspath>${project.build.directory}/jakarta.data-api.jar
+                        </signature.sigTestClasspath>
+                        <java.util.logging.config.file>target/test-classes/logging.properties
+                        </java.util.logging.config.file>
+                        <maven.repo.local>${settings.localRepository}</maven.repo.local>
+                        <jboss.home>${jboss.home}</jboss.home>
+                        <arquillian.launch>wildfly-core-profile</arquillian.launch>
+                        <jakarta.tck.skip.deployment>false</jakarta.tck.skip.deployment>
+                        <!-- This property propagates to the javaVmArguments in arquillian.xml -->
+                        <additional.vm.args>
+                        </additional.vm.args>
+                    </systemPropertyVariables>
+                    <environmentVariables>
+                        <JBOSS_HOME>${jboss.home}</JBOSS_HOME>
+                    </environmentVariables>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
+    <profiles>
+        <!-- Provision wildfly to target/wildfly if jboss.home isn't externally configured -->
+        <profile>
+            <id>install-wildfly</id>
+            <activation>
+                <property>
+                    <name>!jboss.home</name>
+                </property>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.wildfly.plugins</groupId>
+                        <artifactId>wildfly-maven-plugin</artifactId>
+                        <version>${version.org.wildfly.maven.plugin}</version>
+                        <executions>
+                            <execution>
+                                <id>install-wildfly</id>
+                                <phase>initialize</phase>
+                                <goals>
+                                    <goal>provision</goal>
+                                </goals>
+                            </execution>
+                        </executions>
+                        <configuration>
+                            <provisioningDir>${jboss.home}</provisioningDir>
+                            <overwrite-provisioned-server>true</overwrite-provisioned-server>
+                            <feature-packs>
+                                <feature-pack>
+                                    <groupId>${wildfly.feature.pack.groupId}</groupId>
+                                    <artifactId>${wildfly.feature.pack.artifactId}</artifactId>
+                                    <version>${version.org.wildfly.wildfly}</version>
```

Comment:
FWIW if we leave the `version.org.wildfly.wildfly` property empty and use a channel, we don't have to upgrade the version for every new release.

Reply:
@jamezp I'd prefer to leave this for the future. I want this to work with all three of wildfly, wildfly-ee and wildfly-preview feature packs and I don't have time now to incorporate channel wiring and test the combinations.